genai-perf/genai_perf/inputs/input_retriever_factory.py

```diff
@@ -62,44 +59,21 @@ def get_input_data(self) -> Dict:
             The generic dataset JSON
         """
 
-        if self.config.output_format == OutputFormat.OPENAI_EMBEDDINGS:
-            if self.config.input_type != PromptSource.FILE:
-                raise GenAIPerfException(
-                    f"{OutputFormat.OPENAI_EMBEDDINGS.to_lowercase()} only supports a file as input."
-                )
-            input_file_dataset = self._get_input_dataset_from_embeddings_file()
-            generic_dataset_json = (
-                self._convert_input_synthetic_or_file_dataset_to_generic_json(
-                    input_file_dataset
-                )
-            )
-        elif self.config.output_format == OutputFormat.RANKINGS:
-            if self.config.input_type != PromptSource.FILE:
-                raise GenAIPerfException(
-                    f"{OutputFormat.RANKINGS.to_lowercase()} only supports a directory as input."
-                )
-            queries_filename = self.config.input_filename / "queries.jsonl"
-            passages_filename = self.config.input_filename / "passages.jsonl"
-            input_file_dataset = self._get_input_dataset_from_rankings_files(
-                queries_filename, passages_filename
-            )
-            generic_dataset_json = (
-                self._convert_input_synthetic_or_file_dataset_to_generic_json(
-                    input_file_dataset
-                )
-            )
-        elif self.config.output_format == OutputFormat.IMAGE_RETRIEVAL:
-            if self.config.input_type != PromptSource.FILE:
-                raise GenAIPerfException(
-                    f"{OutputFormat.IMAGE_RETRIEVAL.to_lowercase()} only supports a file as input."
-                )
-            input_file_dataset = self._get_input_dataset_from_file()
-            input_file_dataset = self._encode_images_in_input_dataset(
-                input_file_dataset
-            )
+        if self.config.output_format in [
+            OutputFormat.OPENAI_EMBEDDINGS,
+            OutputFormat.RANKINGS,
+            OutputFormat.IMAGE_RETRIEVAL,
+        ]:
+            # TODO: remove once the factory fully integrates retrievers
```

Comment:
Remove what here? Does this mean you're planning to have a factory that creates the correct retriever based on the input source in the future?

*If so, my comment above about a base retriever class can be resolved, since you're already planning to go that route.

Reply:
You are correct 😄  We want the factory to handle all the retrievers in the future based on the input source, but that requires all the retrievers to be implemented. Consider this change as a step towards that direction without breaking the current overall behavior.

Reply:
Perfect, thanks for clarifying! That sounds excellent.